modules/core/src/arithm.cpp

```diff
@@ -947,7 +947,25 @@ void cv::copyTo(InputArray _src, OutputArray _dst, InputArray _mask)
 namespace cv
 {
 
-static BinaryFuncC* getMulTab()
+static void mul8u16uWrapper(const uchar* src1, size_t step1,
+                               const uchar* src2, size_t step2,
+                               uchar* dst, size_t step, int width, int height,
+                               void* usrdata)
+{
+    double scale = *((double*)usrdata);
+    cv_hal_mul8u16u(src1, step1, src2, step2, (ushort*)dst, step, width, height, scale);
+}
+
+static void mul8s16sWrapper(const uchar* src1, size_t step1,
+                               const uchar* src2, size_t step2,
+                               uchar* dst, size_t step, int width, int height,
+                               void* usrdata)
+{
+    double scale = *((double*)usrdata);
+    cv_hal_mul8s16s((schar*)src1, step1, (schar*)src2, step2, (short*)dst, step, width, height, scale);
```

Comment:
`cv_hal_mul8s16s` and `cv_hal_mul8u16u` return value is not checked. They may return error or not implemented status.

Reply:
Have an ugly solution for this, please take a look